### PR TITLE
Intelligent Service JourneyAI's output schema.

### PR DESCRIPTION
**Pull request by daiadobe**
JourneyAI intelligent service is designed to provide more insights and recommendations for message deliveries through both email channel and push channel. Currently, it's available for Adobe Campaign Standard (intro [here](https://experienceleague.adobe.com/docs/campaign-standard/using/testing-and-sending/preparing-and-testing-messages/predictive.html?lang=en#journey-ai)). The service provides engagement scores and send time optimizations, both as profile extensions. All outputs intend to be under **XDM Individual Profile._experience** with detailed structures shown below.
- Engagement Scores
image
- Send Time Optimization
image

### CODE REVIEW

**Comment by mjasrotia** [comment]
@daiadobe , Can we please set up a quick meeting to understand the intent here ? Please do include our Product Manager (Lakshmi Shivalingaiah) in that meeting too. A few observations here :
1. Seems like you are putting together a bunch of schemas under your own solution specific repository. 
2. I see you are still using the "xdm" namespace to define properties. If they are defined as standard set of attributes, should they be defined as core set of schemas ?

We can discuss more during our meeting.

**Comment by daiadobe** [comment]
> @daiadobe , Can we please set up a quick meeting to understand the intent here ? Please do include our Product Manager (Lakshmi Shivalingaiah) in that meeting too. A few observations here :
> 
> 1. Seems like you are putting together a bunch of schemas under your own solution specific repository.
> 2. I see you are still using the "xdm" namespace to define properties. If they are defined as standard set of attributes, should they be defined as core set of schemas ?
> 
> We can discuss more during our meeting.

1. yes. Intelligent Services will be a set of services on AEP and JourneyAI will be one of them.
2. Not fully understand the difference but will explain our intent during the discussion.

Will send the meeting invite shortly

**Comment by daiadobe** [comment]
@mjasrotia Hi Manik, I've updated this PR following our discussion. Please let me know if there's any further comments.
Thanks,

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "$id": "https://ns.adobe.com/experience/intelligentServices/profile-journeyai-engagementscores",
+  "title": "Intelligent Services JourneyAI Engagement Scores",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:extends": ["https://ns.adobe.com/xdm/context/profile"],
```
Comment (mjasrotia) [inline extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json:14]: "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/profile"]

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            }
+          },
+          "meta:xdmType": "object"
+        }
+      }
+    }
+  },
+  "allOf": [
+    {
+      "$ref": "https://ns.adobe.com/xdm/context/profile",
```
Comment (mjasrotia) [inline extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json:93]: Please Remove this

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                  "title": "emailEngagementScores",
+                  "description": "Engagement scores for email channel messages",
+                  "type": "object",
+                  "isRequired": false,
+                  "properties": {
+                    "https://ns.adobe.com/experience/intelligentServices/journeyAI/emailClickEngagementScore": {
+                      "title": "emailClickEngagementScore",
+                      "description": "probability for a customer to click on links in an email",
+                      "type": "number",
+                      "isRequired": false,
```
Comment (mjasrotia) [inline extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json:41]: We don't use isRequired flag at the property level. Please remove all across

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json [hunk 4]
```diff
@@ -0,0 +1,103 @@
+{
+  "meta:license": [
+    "Copyright 2018 Adobe Systems Incorporated. All rights reserved.",
```
Comment (mjasrotia) [inline extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json:3]: 2020

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                      "type": "number",
+                      "isRequired": false,
+                      "meta:xdmType": "number"
+                    },
+                    "https://ns.adobe.com/experience/intelligentServices/journeyAI/emailOpenEnagementScore": {
+                      "title": "emailOpenEnagementScore",
+                      "description": "probability of a customer to open an email",
+                      "type": "number",
+                      "isRequired": false,
+                      "meta:xdmType": "number"
```
Comment (mjasrotia) [inline extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json:49]: we can remove "meta:xdmType" all across

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+                      "description": "the probability for recipient to unsubscribe from email channel given one email opened",
+                      "type": "number",
+                      "isRequired": false,
+                      "meta:xdmType": "number"
+                    }
+                  },
+                  "meta:xdmType": "object"
+                },
+                "https://ns.adobe.com/experience/intelligentServices/journeyAI/engagementScoresInfo": {
+                  "title": "engagementScoresInfo",
```
Comment (mjasrotia) [inline extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json:76]: Engagement Scores Info (No Period "." in the end)

Please change all the titles

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+                    "https://ns.adobe.com/experience/intelligentServices/journeyAI/emailRetentionPercentRank": {
+                      "title": "emailRetentionPercentRank",
+                      "description": "percentile rank of the likelihood of staying subscribed to the email list",
+                      "type": "number",
+                      "isRequired": false,
+                      "meta:xdmType": "number"
+                    },
+                    "https://ns.adobe.com/experience/intelligentServices/journeyAI/emailUnsubscriptionProbability": {
+                      "title": "emailUnsubscriptionProbability",
+                      "description": "the probability for recipient to unsubscribe from email channel given one email opened",
```
Comment (mjasrotia) [inline extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json:67]: Upper case the first letter. Ends with a "." period

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+  },
+  "allOf": [
+    {
+      "$ref": "https://ns.adobe.com/xdm/context/profile",
+      "type": "object",
+      "meta:xdmType": "object"
+    },
+    {
+      "$ref": "#/definitions/journeyai-engagement-scores",
+      "type": "object",
```
Comment (mjasrotia) [inline extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json:99]: Remove "type" and "metaType"

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json (hunk 1 above)
Comment (daiadobe) [reply extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json]: Done

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json (hunk 2 above)
Comment (daiadobe) [reply extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json]: Done

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json (hunk 3 above)
Comment (daiadobe) [reply extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json]: Done

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json (hunk 4 above)
Comment (daiadobe) [reply extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json]: Done

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json (hunk 5 above)
Comment (daiadobe) [reply extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json]: Done

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json (hunk 6 above)
Comment (daiadobe) [reply extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json]: Done

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json (hunk 7 above)
Comment (daiadobe) [reply extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json]: Done

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json (hunk 8 above)
Comment (daiadobe) [reply extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json]: Done

**Comment by daiadobe** [comment]
@mjasrotia Hi Manik, thank you for all the comments. I've accommodated all of them. Please let me know if there's any further comments.
Thanks,

File: extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+          }
+        }
+      }
+    }
+  },
+  "allOf": [
+    {
+      "$ref": "#/definitions/journeyai-engagement-scores"
+    }
+  ]
```
Comment (mjasrotia) [inline extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json:77]: Please add "meta:status": "experimental". Once it is good to go to PROD, we can change the status to "stable".. Please add this to the other schema too.

Comment (daiadobe) [reply extensions/adobe/experience/intelligentServices/profile-journeyai-engagementscores.schema.json]: Done @mjasrotia. Thanks,

**Comment by prabhum2** [comment]
retest

**Review by mjasrotia — APPROVED** [review]

**Review by prabhum2 — APPROVED** [review]

**Pull request merged**